java-squid/src/main/java/org/sonar/java/ast/parser/JavaGrammar.java

```diff
@@ -164,8 +164,8 @@ public ImportClauseTree IMPORT_DECLARATION() {
           f.newEmptyImport(b.invokeRule(JavaPunctuator.SEMI))));
   }
 
-  public AstNode TYPE_DECLARATION() {
-    return b.<AstNode>nonterminal(JavaLexer.TYPE_DECLARATION)
+  public Tree TYPE_DECLARATION() {
+    return b.<Tree>nonterminal(JavaLexer.TYPE_DECLARATION)
       .is(
         b.firstOf(
           // TODO Unfactor MODIFIERS? It always seems to precede CLASS_DECLARATION()
```

Comment:
![INFO](https://raw.githubusercontent.com/henryju/image-hosting/master/info.png)Complete the task associated to this TODO comment.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)